src/generic_ui/polymer/root.ts

```diff
@@ -90,17 +92,60 @@ Polymer({
     // to sync the value to core
     core.updateGlobalSettings(model.globalSettings);
   },
-  showToast: function(e, data) {
-    this.$.toast.setAttribute('text', data.text);
-    this.$.toast.show();
-  },
   signalToFireChanged: function() {
     if (this.ui.signalToFire != '') {
       this.fire('core-signal', {name: this.ui.signalToFire});
       this.ui.signalToFire = '';
     }
   },
-  observe: {
-    'ui.signalToFire' : 'signalToFireChanged'
+  /* All functions below help manage paper-toast behaviour. */
+  showToast: function(e, data) {
+    this.ui.toastMessage = data.text;
+  },
+  closeToast: function() {
+    this.ui.toastMessage = null;
+  },
+  messageNotNull: function(toastMessage) {
+    // Whether the toast is shown is controlled by if ui.toastMessage
+    // is null. This function returns whether ui.toastMessage == null,
+    // and also sets a timeout to close the toast.
+    if (toastMessage) {
+      clearTimeout(this.clearToastTimeout);
+      this.clearToastTimeout = setTimeout(this.closeToast, 10000);
+      return true;
+    }
+    return false;
+  },
+  openTroubleshoot: function() {
+    if (this.stringMatches(ui.toastMessage, UI.GET_FAILED_MSG)) {
+      this.troubleshootTitle = "Unable to get access";
+    } else {
+      this.troubleshootTitle = "Unable to share access";
+    }
+    this.closeToast();
+    this.fire('core-signal', {name: 'open-troubleshoot'});
+  },
+  stringMatches: function(toastMessage, failureMsgConstant) {
+    // Determine if the error in the toast is a getter or sharer error
+    // by comparing the error string to getter/sharer error constants.
+    if (toastMessage) {
+      return toastMessage.indexOf(failureMsgConstant) > -1;
+    }
+    return false;
+  },
+  topOfStatuses: function(gettingStatus, sharingStatus) {
```

Comment:
Is there a way that we can put the toast inside of the roster, and not have to worry about setting it's position?  Feel free to make this a TODO..